app/src/main/java/com/example/inputcontrols042homework/MainActivity.java

```diff
@@ -20,31 +20,34 @@ protected void onCreate(Bundle savedInstanceState) {
 
         showToast = findViewById(R.id.show_toast_Button);
         showToast.setOnClickListener(this);
-        checkBoxes = new CheckBox[]{findViewById(R.id.chocolate_syrup_checkBox),
+        checkBoxes = new CheckBox[]{
+                findViewById(R.id.chocolate_syrup_checkBox),
                 findViewById(R.id.sprinkles_checkBox),
                 findViewById(R.id.crushed_nuts_checkBox),
                 findViewById(R.id.cremes_checkBox),
-                findViewById(R.id.orio_cookie_crumbles_checkBox)};
+                findViewById(R.id.orio_cookie_crumbles_checkBox)
+        };
     }
 
     public String getToastMessage() {
         String message = "Toppings: ";
+        String control = null;
         for (CheckBox checkBox : checkBoxes) {
             if (checkBox.isChecked()) {
-                message = message.concat(checkBox.getText().toString() + ", ");
+                if (control == null) {
```

Comment:
Flag variable of type String is bad idea. It will be better to create another variable `String products = null`;
Use it instead of control and message:
```if(products == null) {
    products.concat(checkBox.getText().toString());
} else {
    products.concat(", " + checkBox.getText().toString());
}